TestsCalculator/CalculatorPageTests.cs

```diff
@@ -164,14 +427,34 @@ public void CheckYearInDropdown()
         {
             //Arrange
             IWebElement yearDropdown = driver.FindElement(By.Id("year"));
-            //Act
-            //object of SelectElement
+
+            //Act & Assert
             SelectElement s = new SelectElement(yearDropdown);
             //get all options
             IList<IWebElement> els = s.Options;
             //count options
             int e = els.Count;
             Assert.Greater(e, 15);
         }
+
+        [Test]
+        public void SelectFeb29LeapYear()
+        {
+            //Arrange
+            IWebElement termField = driver.FindElement(By.Id("term"));
+            IWebElement dayDropdown = driver.FindElement(By.Id("day"));
+            IWebElement monthDropdown = driver.FindElement(By.Id("month"));
+            IWebElement yearDropdown = driver.FindElement(By.Id("year"));
+
+            //Act
+            termField.SendKeys("1");
+            dayDropdown.SendKeys("29"); 
+            monthDropdown.SendKeys("February");
+            yearDropdown.SendKeys("2024");
+
+            //Assert
+            IWebElement endDateField = driver.FindElement(By.Id("endDate"));
+            Assert.AreEqual("01/03/2024", endDateField.GetAttribute("value"), "Date is incorrect");
```

Comment:
Let's discuss, there are few other options for this test:

```suggestion
            SelectElement s = new SelectElement(dayDropdown);
            IList<string> actualDays = new List<string>();
            for (int j = 0; j < s.Options.Count; j++)
            {
                    actualDays.Add(s.Options.ElementAt(j).Text);
            }
            
            IList<string> expectedDays = new List<string>();
            for (int j = 0; j < 29; j++)
            {
                    expectedDays.Add(j.ToString());
            }
            
            Assert.Contains("29", actualDays);
            Assert.AreEqual(29, s.Options.Count);
            Assert.AreEqual(expectedDays, actualDays);
```